src/services/expensesService.js

```diff
@@ -0,0 +1,54 @@
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
```

Comment:
You are using multiple filter functions here. While this works, it is not efficient because each filter function will iterate over the entire array. Consider combining these filters into a single filter function to improve performance.